framework/src/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java

```diff
@@ -1242,15 +1242,19 @@ public AnnotationMirror createIntRangeAnnotation(long from, long to) {
     }
 
     /**
-     * Create an {@code @IntRange} annotation from the range. May return BOTTOMVAL or UNKNOWNVAL.
+     * Create an {@code @IntRange} or {@code @IntVal} annotation from the range. May return
+     * BOTTOMVAL or UNKNOWNVAL.
      */
     public AnnotationMirror createIntRangeAnnotation(Range range) {
         if (range.isNothing()) {
             return BOTTOMVAL;
         } else if (range.isEverything()) {
             return UNKNOWNVAL;
-        } else {
+        } else if (range.isWiderThan(MAX_VALUES)) {
             return createIntRangeAnnotation(range.from, range.to);
+        } else {
```

Comment:
I had thought that, even though some annotations are implicit (computed by dataflow) and some have been converted, it was still better to retain the programmer-written form when possible -- that is, not to gratuitously change them.

Can you remind me the benefit of canonicalizing the annotations into a consistent form here?